### DISCUSSION TITLE
Data security/privacy when using pyannote diarization on Huggingface

### DISCUSSION THREAD

**Discussion by virtualarchitectures** [pyannote/pyannote-audio#1401] [Q&A] [answered]
Hi,

I'm working on a basic tool for interview transcription. I'd like to use pyannote for diarization but I'd like to understand what information is sent to the Huggingface servers when pyannote calls their API. I'm aware of Huggingface's docuementation on security (https://huggingface.co/docs/hub/security) and security and compliance (https://huggingface.co/docs/inference-endpoints/security#data-securityprivacy). This question is to better understand what information is being transacted. What's in the payload and what comes back. 

Many thanks.

**Comment by hbredin** [comment, marked as answer]
pyannote can be used independently of Huggingface.

See ["Can I use gated models and pipelines offline?"](https://github.com/pyannote/pyannote-audio/blob/develop/FAQ.md#can-i-use-gated-models-and-pipelines-offline) FAQ.

**Reply by virtualarchitectures** [reply]
Ace! Thanks for the speedy reply. I was looking for a Wiki and missed the FAQ doc. That's great. Many thanks for this.

**Reply by infoseekAI** [reply]
the links in the FAQ are broken. any chance you can update them?

e.g. https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/applying_a_model.ipynb (error occurred)

**Comment by virtualarchitectures** [comment]
Hi @hbredin your guide was really good and I have it running. However, my pipeline seems to run smoother when working with the Huggingface connection. For reference I'm using it with WhisperX. Could you please clarify what the Huggingface connection does? Is just authenticating use and providing downloads of the model's and pipelines or is any of the text I'm processing being sent over the connection to huggingface's servers? Many thanks.

**Reply by hbredin** [reply]
No data leaves your computer when calling `diarization = pipeline("audio.wav")`.

**Reply by timlac** [reply]
This might be a stupid question, but can you confirm that no data leaves my computer when running the voice activity detection pipeline either? https://huggingface.co/pyannote/voice-activity-detection

**Reply by hbredin** [reply]
Yes. 

But may I suggest not to trust blindly the words of a random stranger in an internet forum? 
I do [consulting](https://herve.niderb.fr/consulting.html) too...

**Comment by timlac** [comment]
I have the same question. What exactly does the Huggingface connection do? Is any of my audio data sent to the servers?
